File: keras.preprocessing_text.py
import sys
import json
from keras.preprocessing.text import Tokenizer

# Get the values from sys.argv
arguments = sys.argv[1:]  # Exclude the script name from the list
num_words = None
oov_token = None

for arg in arguments:
    if arg.startswith('num_words='):
        num_words = int(arg.split('=')[1])
    elif arg.startswith('oov_token='):
        oov_token = arg.split('=')[1]

# Create the Tokenizer instance
tokenizer = Tokenizer(num_words=num_words, oov_token=oov_token)

for line in sys.stdin:
    split_string = line.split(':::') # use triple separator
    if split_string[0] == 'fit_on_texts':
        parsed_json = json.loads(split_string[1])
        tokenizer.fit_on_texts(parsed_json)
    if split_string[0] == 'word_index':
        json_word_index = json.dumps(tokenizer.word_index)
        sys.stdout.write(json_word_index)
    if split_string[0] == 'texts_to_sequences':
        parsed_json = json.loads(split_string[1])
        text_sequences = tokenizer.texts_to_sequences(parsed_json)
        json_text_sequences = json.dumps(text_sequences)
        sys.stdout.write(json_text_sequences)

File: keras_preprocessing_text.js
/**
 * @license
 * Copyright 2019 Shahrul Nizam Selamat All Rights Reserved.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * =============================================================================
 */

import { spawn, exec } from 'child_process'
import path, { dirname } from 'path'
import { fileURLToPath } from 'url'

const __filename = fileURLToPath(import.meta.url)
const __dirname = dirname(__filename)

const pyPath = path.resolve(__dirname, 'keras.preprocessing_text.py')

function onData (data) {
  return function (resolve) {
    let output
    try {
      output = JSON.parse(data)
    } catch (e) {
      output = data.toString()
    }
    resolve(output)
    this.pyp.stdout.removeListener('data', onData)
  }.bind(this)
}

export default class Tokenizer {
  constructor (num_words, oov_token) {
    this.num_words = num_words
    this.oov_token = oov_token
    const args = []
    if(num_words){
      args.push(`num_words=${num_words}`)
    }
    if(oov_token){
      args.push(`oov_token=${oov_token}`)
    }
    this.pyp = spawn('python', ['-u', pyPath, ...args])

    this.pyp.stderr.on('data', (data) => {
      console.log(`Error: ${data}`)
    })

    this.pyp.on('close', () => {})
  }

  async fit_on_texts (vectors) {
    this.pyp.stdin.write(`fit_on_texts:::${JSON.stringify(vectors)}\n`)
  }

  async word_index () {
    return new Promise(resolve => {
      this.pyp.stdout.on('data', data => onData.call(this, data)(resolve))
      this.pyp.stdin.write('word_index:::\n')
    })
  }

  async texts_to_sequences (vectors) {
    return new Promise(resolve => {
      this.pyp.stdout.on('data', data => onData.call(this, data)(resolve))
      this.pyp.stdin.write(`texts_to_sequences:::${JSON.stringify(vectors)}\n`)
    })
  }

  close () {
    const pid = this.pyp.pid
    const killCommand = `kill ${pid}`
    exec(killCommand, (killError, killStdout, killStderr) => {
      if (killError) {
        this.pyp.kill()
      }
    })
  }
}
